backend/app/deployment_handler.go

```diff
@@ -722,3 +775,248 @@ func (h *Handler) HandleRemoveNode(c *gin.Context) {
+	vms, err := h.db.ListUserVMS(id)
+	if err != nil {
+		logger.GetLogger().Error().Err(err).Str("user_id", id).Msg("Failed to list user VMs")
+		InternalServerError(c)
+		return
+	}
+
+	vmList := make([]VMItem, 0, len(vms))
+	for _, vm := range vms {
+		vmResult, err := vm.GetVMResult()
+		if err != nil {
+			logger.GetLogger().Error().Err(err).Int("vm_id", vm.ID).Msg("Failed to deserialize VM result")
+			continue
+		}
+
+		vmList = append(vmList, VMItem{
+			ID:          vm.ID,
+			ProjectName: vm.ProjectName,
+			VM:          vmResult,
+			CreatedAt:   vm.CreatedAt,
+		})
+	}
+
+	Success(c, http.StatusOK, "Vms are listed successfully", ListVMsResponse{
+		VMs:   vmList,
+		Count: len(vmList),
+	})
+}
+
+// @Summary List VM
+// @Description Get specific VM for authenticated user
+// @Tags vms
+// @Security BearerAuth
+// @Accept json
+// @Produce json
+// @Param id path string true "VM ID"
+// @Success 200 {object} ListVMResponse
+// @Failure 400 {object} APIResponse "Bad Request"
+// @Failure 401 {object} APIResponse "Unauthorized"
+// @Failure 404 {object} APIResponse "VM not found"
+// @Failure 500 {object} APIResponse "Internal server error"
+// @Router /deployments/vms/{id} [get]
+func (h *Handler) HandleListVM(c *gin.Context) {
+	config, err := h.getClientConfig(c)
+	if err != nil {
+		InternalServerError(c)
+		return
+	}
+
+	vmID := c.Param("id")
+	if vmID == "" {
+		Error(c, http.StatusBadRequest, "Vm id is required", "")
+		return
+	}
+
+	vm, err := h.db.GetVMByID(config.UserID, vmID)
+	if err != nil {
+		logger.GetLogger().Error().Err(err).
+			Str("user_id", config.UserID).
+			Str("vm_id", vmID).
+			Msg("Failed to find VM")
+		Error(c, http.StatusNotFound, "Vm is not found", err.Error())
+		return
+	}
+
+	vmResult, err := vm.GetVMResult()
+	if err != nil {
+		logger.GetLogger().Error().Err(err).
+			Int("vm_id", vm.ID).
+			Msg("Failed to deserialize VM result")
+		InternalServerError(c)
+		return
+	}
+
+	Success(c, http.StatusOK, "Vm is listed successfully", ListVMResponse{VM: VMItem{
+		ID:          vm.ID,
+		ProjectName: vm.ProjectName,
+		VM:          vmResult,
+		CreatedAt:   vm.CreatedAt,
+	}})
+}
+
+// @Summary Delete VM
+// @Description Delete a VM for authenticated user
+// @Tags vms
+// @Security BearerAuth
+// @Accept json
+// @Produce json
+// @Param id path string true "VM ID"
+// @Success 200 {object} DeleteVMResponse "Deletion workflow started"
+// @Failure 400 {object} APIResponse "Bad Request"
+// @Failure 401 {object} APIResponse "Unauthorized"
+// @Failure 404 {object} APIResponse "VM not found"
+// @Failure 500 {object} APIResponse "Internal server error"
+// @Router /deployments/vms/{id} [delete]
+func (h *Handler) HandleDeleteVM(c *gin.Context) {
+	config, err := h.getClientConfig(c)
+	if err != nil {
+		InternalServerError(c)
+		return
+	}
+
+	vmID := c.Param("id")
+	if vmID == "" {
+		Error(c, http.StatusBadRequest, "vm id is required", "")
+		return
+	}
+	vm, err := h.db.GetVMByID(config.UserID, vmID)
```

Comment:
We should check the access on a higher level not the database level, this works, but can be risky.

should be something like authorizationService.CanAccessVM(userId, vmID), before trying to do any. operations